receiver/splunkenterprisereceiver/client_test.go

```diff
@@ -4,148 +4,179 @@
-				data := strings.NewReader("example search")
-				req, _ := http.NewRequest(method, url, data)
-				req.Header.Add("Authorization", client.basicAuth)
-				req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
-				return req
-			}(),
-		},
-		{
-			desc: "Second req, jobID detected",
-			sr: &searchResponse{
-				search: "example search",
-				Jobid:  &testJobID,
-			},
-			client: client,
-			expected: func() *http.Request {
-				method := "GET"
-				path := fmt.Sprintf("/services/search/jobs/%s/results", testJobID)
-				testEndpoint, _ := url.Parse("https://localhost:8089")
-				url, _ := url.JoinPath(testEndpoint.String(), path)
-				req, _ := http.NewRequest(method, url, nil)
-				req.Header.Add("Authorization", client.basicAuth)
-				req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
-				return req
-			}(),
+			Timeout:            11 * time.Second,
 		},
 	}
 
-	ctx := context.Background()
-	for _, test := range tests {
-		t.Run(test.desc, func(t *testing.T) {
-			req, err := test.client.createRequest(ctx, test.sr)
-			require.NoError(t, err)
-			// have to test specific parts since individual fields are pointers
-			require.Equal(t, test.expected.URL, req.URL)
-			require.Equal(t, test.expected.Method, req.Method)
-			require.Equal(t, test.expected.Header, req.Header)
-			require.Equal(t, test.expected.Body, req.Body)
-		})
-	}
-}
-
-// createAPIRequest creates a request for api calls i.e. to introspection endpoint
-func TestAPIRequestCreate(t *testing.T) {
-	client := newSplunkEntClient(&Config{
-		Username:          "admin",
-		Password:          "securityFirst",
-		MaxSearchWaitTime: 11 * time.Second,
-		HTTPClientSettings: confighttp.HTTPClientSettings{
-			Endpoint: "https://localhost:8089",
+	host := &mockHost{
+		extensions: map[component.ID]component.Component{
+			component.NewID("basicauth/client"): auth.NewClient(),
 		},
-		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
-			CollectionInterval: 10 * time.Second,
-			InitialDelay:       1 * time.Second,
-		},
-	})
-
-	ctx := context.Background()
-	req, err := client.createAPIRequest(ctx, "/test/endpoint")
+	}
+	// create a client from an example config
+	client, err := newSplunkEntClient(cfg, host, componenttest.NewNopTelemetrySettings())
 	require.NoError(t, err)
 
-	expectedURL := client.endpoint.String() + "/test/endpoint"
-	expected, _ := http.NewRequest(http.MethodGet, expectedURL, nil)
-	expected.Header.Add("Authorization", client.basicAuth)
-	expected.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	testEndpoint, _ := url.Parse("https://localhost:8089")
 
-	require.Equal(t, expected.URL, req.URL)
-	require.Equal(t, expected.Method, req.Method)
-	require.Equal(t, expected.Header, req.Header)
-	require.Equal(t, expected.Body, req.Body)
+	require.Equal(t, client.endpoint, testEndpoint)
 }
+
+// test functionality of createRequest which is used for building metrics out of
```

Comment:
just push this to a separate branch so it's not merged with commented out code, the code will otherwise rot very quickly

Reply:
I will do you one better and just fix the tests! Sorry about that